### PR TITLE
Update 110_talks.rst

### LINKED ISSUE #4263: Update for https://restic.readthedocs.io/en/stable/110_talks.html
this page:
https://restic.readthedocs.io/en/stable/110_talks.html

is slightly outdated from this page:
https://restic.net/#talks

so next i want to create a pull request for my updates to add
"2021-03-21: [The Changelog: Restic has your backup (Podcast)](https://changelog.com/podcast/434)"

### PR DESCRIPTION
**Pull request by GuitarBilly**
as described

What does this PR change? What problem does it solve?
-----------------------------------------------------
update https://restic.readthedocs.io/en/stable/110_talks.html

Was the change previously discussed in an issue or on the forum?
----------------------------------------------------------------
Closes #4263 

Checklist
---------

- [X] I have read the [contribution guidelines](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#providing-patches).
- [X] I have [enabled maintainer edits](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added tests for all code changes.
- [ ] I have added documentation for relevant changes (in the manual).
- [ ] There's a new file in `changelog/unreleased/` that describes the changes for our users (see [template](https://github.com/restic/restic/blob/master/changelog/TEMPLATE)).
- [ ] I have run `gofmt` on the code in all commits.
- [ ] All commit messages are formatted in the same style as [the other commits in the repo](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#git-commits).
- [X] I'm done! This pull request is ready for review.

### CODE REVIEW

File: doc/110_talks.rst [hunk 1]
```diff
@@ -17,6 +17,8 @@ Talks
 
 The following talks will be or have been given about restic:
 
+-  2021-03-21: `The Changelog: Restic has your backup
```
Comment (rawtaz) [inline doc/110_talks.rst:20]: Where did you get this date from? According to the Changelog page for this podcast it was recorded on the 10th of March and published no the 2nd of April, neither of which match :)

Comment (GuitarBilly) [reply doc/110_talks.rst:20]: As listed in the issue https://github.com/restic/restic/issues/4263 that i created I took it litterally from 
https://restic.net/#talks where it prints:
2021-03-21: [The Changelog: Restic has your backup (Podcast)](https://changelog.com/podcast/434)


And you are right, the podcast page https://changelog.com/podcast/434 mentions other dates. I can only guess above date of 21st March is when the restic.net page was updated.

I would propose to make it the date it was published as that is what you see in the podcast list and podcast app, being 2nd of April 2021.
https://changelog.com/podcast?page=5#feed

And I would propose to align the https://restic.net/#talks date for this entry.


Can you agree?

Comment (rawtaz) [reply doc/110_talks.rst:20]: Ah, that makes sense. Yeah, that is the date that the page on the website was published. I agree with you, better to use the 2nd of April 2021 as the date in your PR :) Can you make that change and amend/squash the commit?

I can align the date, doesn't hurt.

Comment (rawtaz) [reply doc/110_talks.rst]: You did!

**Review by rawtaz — APPROVED** [review]
[APPROVED] <-- (stamp)

**Pull request merged**